comp/serializer/compression/mock/component_mock.go

```diff
@@ -5,9 +5,12 @@
 
 //go:build test
 
-package compression
+// Package mock provides the mock component for serializer/compression
+package mock
```

Comment:
Looking at the code it seems that we are not using the Mock interface at all. Everything delegates to the `noopimpl` Shoudl we consolidate everything into a single file?

Reply:
That's correct - I included it because the component architecture guidelines said i must have it.  Do you want me to change it?

Reply:
Let's leave it for now and see if we can refactor things on a separate PR